lib/utils.js

```diff
@@ -181,6 +187,117 @@ exports.getFunctionNodeFromIIFE = function(node) {
 };
 
 /**
+ * Returns the type of AST node or ECMAScript production in which the provided
+ * open parenthesis token is included.
+ *
+ * @param {Object} token
+ * @param {JsFile} file
+ * @returns {String}
+ */
+exports.categorizeOpenParen = function(token, file) {
```

Comment:
I'm not particularly fond of adding more function to utils. At the very least, these should be functions dangled on the rule export unless other rules will use them. If they're going to be reused, then these functions could be properties of a `TokenCategorizer` class that's within `lib`.